src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -73,4 +73,66 @@ private void validDistance(Long newDistance, Long exitDistance) {
     private void addSection(Section section) {
         sections.add(section);
     }
+
+    public void delete(Station station) {
+        validSectionsSizeCheck();
+        Section upSection = sections.stream()
+                .filter(section -> section.getDownStation().equals(station))
+                .findFirst()
+                .orElse(Section.empty());
+
+        Section downSection = sections.stream()
+                .filter(section -> section.getUpStation().equals(station))
+                .findFirst()
+                .orElse(Section.empty());
+
+        if (isValidAndEndStation(upSection, downSection)) {
+            createSection(upSection, downSection);
+        }
+
+        updateUpStationOrDownStation(upSection, downSection);
+        removeSection(upSection, downSection);
+    }
+
+    private void validSectionsSizeCheck() {
+        if (sections.size() <= 1) {
+            throw new IllegalArgumentException("구간이 하나인 노선인 경우 구간 삭제 할 수 없습니다.");
+        }
+    }
+
+    private boolean isValidAndEndStation(Section upSection, Section downSection) {
+        if (upSection.isEmpty() && downSection.isEmpty()) {
+            throw new IllegalArgumentException("노선에 등록된 역이 아닙니다.");
+        }
+
+        if (upSection.isEmpty() || downSection.isEmpty()) {
+            return false;
+        }
+
+        return true;
+    }
```

Comment:
다음과 같이 해볼 수 있을 것 같아요 😄 
```suggestion
    private boolean isValidAndEndStation(Section upSection, Section downSection) {
        if (upSection.isEmpty() && downSection.isEmpty()) {
            throw new IllegalArgumentException("노선에 등록된 역이 아닙니다.");
        }

        return !upSection.isEmpty() && !downSection.isEmpty();
    }
```

Reply:
바로 return 되도록 수정하였습니다!